diff -ruN OpenFDTD.org/mpi/Makefile OpenFDTD/mpi/Makefile
--- OpenFDTD.org/mpi/Makefile	2019-06-08 13:39:10.000000000 +0900
+++ OpenFDTD/mpi/Makefile	2019-08-09 11:41:39.430843746 +0900
@@ -1,116 +1,117 @@
-PROGRAM = ../ofd_mpi.exe
+PROGRAM = ../ofd_mpi
 
-OMPOPT  = /openmp
-MPIOPT  = /DMPI
+OMPOPT  = -fopenmp
+MPIOPT  = -DMPI
 
-OBJS    =         \
-Main.obj          \
-setupSize.obj     \
-setupSizeNear.obj \
-memallocfree3.obj \
-solve.obj         \
-comm.obj          \
-../src/updateEx.obj       \
-../src/updateEy.obj       \
-../src/updateEz.obj       \
-../src/updateHx.obj       \
-../src/updateHy.obj       \
-../src/updateHz.obj       \
-../src/calcFar1d.obj      \
-../src/calcFar2d.obj      \
-../src/input.obj          \
-../src/monitor.obj        \
-../src/memallocfree1.obj  \
-../src/memallocfree2.obj  \
-../src/setup.obj          \
-../src/setupId.obj        \
-../src/setupNear.obj      \
-../src/setupMurHx.obj     \
-../src/setupMurHy.obj     \
-../src/setupMurHz.obj     \
-../src/setupPmlEx.obj     \
-../src/setupPmlEy.obj     \
-../src/setupPmlEz.obj     \
-../src/setupPmlHx.obj     \
-../src/setupPmlHy.obj     \
-../src/setupPmlHz.obj     \
-../src/setupPml.obj       \
-../src/initfield.obj      \
-../src/vfeed.obj          \
-../src/efeed.obj          \
-../src/epoint.obj         \
-../src/average.obj        \
-../src/murHx.obj          \
-../src/murHy.obj          \
-../src/murHz.obj          \
-../src/pmlEx.obj          \
-../src/pmlEy.obj          \
-../src/pmlEz.obj          \
-../src/pmlHx.obj          \
-../src/pmlHy.obj          \
-../src/pmlHz.obj          \
-../src/dftNear1d.obj      \
-../src/dftNear1dX.obj     \
-../src/dftNear1dY.obj     \
-../src/dftNear1dZ.obj     \
-../src/dftNear2d.obj      \
-../src/dftNear2dX.obj     \
-../src/dftNear2dY.obj     \
-../src/dftNear2dZ.obj     \
-../src/dftNear3d.obj      \
-../src/calcNear3d.obj     \
-../src/nearfield_c.obj    \
-../src/zfeed.obj          \
-../src/spara.obj          \
-../src/outputZfeed.obj    \
-../src/outputSpara.obj    \
-../src/plot3dGeom.obj     \
-../src/writeout.obj       \
-../src/readout.obj        \
-../src/post.obj           \
-../src/farfield.obj       \
-../src/outputFar0d.obj    \
-../src/outputFar1d.obj    \
-../src/outputFar2d.obj    \
-../src/calcNear1d.obj     \
-../src/calcNear2d.obj     \
-../src/logFar1d.obj       \
-../src/logFar2d.obj       \
-../src/logNear1d.obj      \
-../src/logNear2d.obj      \
-../src/plot2dIter.obj     \
-../src/plot2dFreq.obj     \
-../src/plot2dFreq0.obj    \
-../src/plot2dFar1d.obj    \
-../src/plot2dFar1d0.obj   \
-../src/plot3dFar2d.obj    \
-../src/plot3dFar2d0.obj   \
-../src/plot2dNear1d.obj   \
-../src/plot2dNear1d0.obj  \
-../src/plot2d3dNear2d.obj \
-../src/plot2dNear2d0.obj  \
-../src/plot3dNear2d0.obj  \
-../src/plotTools.obj      \
-../src/smithChart.obj     \
-../src/ev2d.obj           \
-../src/ev3d.obj
-
-CC      = cl
-CFLAGS  = /Ox /nologo /I../include /W4 /wd4996
-LDFLAGS = /Ox /nologo
-LIBS    = msmpi.lib
+OBJS    =       \
+Main.o          \
+setupSize.o     \
+setupSizeNear.o \
+memallocfree3.o \
+solve.o         \
+comm.o          \
+../src/updateEx.o       \
+../src/updateEy.o       \
+../src/updateEz.o       \
+../src/updateHx.o       \
+../src/updateHy.o       \
+../src/updateHz.o       \
+../src/calcFar1d.o      \
+../src/calcFar2d.o      \
+../src/input.o          \
+../src/monitor.o        \
+../src/memallocfree1.o  \
+../src/memallocfree2.o  \
+../src/setup.o          \
+../src/setupId.o        \
+../src/setupNear.o      \
+../src/setupMurHx.o     \
+../src/setupMurHy.o     \
+../src/setupMurHz.o     \
+../src/setupPmlEx.o     \
+../src/setupPmlEy.o     \
+../src/setupPmlEz.o     \
+../src/setupPmlHx.o     \
+../src/setupPmlHy.o     \
+../src/setupPmlHz.o     \
+../src/setupPml.o       \
+../src/initfield.o      \
+../src/vfeed.o          \
+../src/efeed.o          \
+../src/epoint.o         \
+../src/average.o        \
+../src/murHx.o          \
+../src/murHy.o          \
+../src/murHz.o          \
+../src/pmlEx.o          \
+../src/pmlEy.o          \
+../src/pmlEz.o          \
+../src/pmlHx.o          \
+../src/pmlHy.o          \
+../src/pmlHz.o          \
+../src/dftNear1d.o      \
+../src/dftNear1dX.o     \
+../src/dftNear1dY.o     \
+../src/dftNear1dZ.o     \
+../src/dftNear2d.o      \
+../src/dftNear2dX.o     \
+../src/dftNear2dY.o     \
+../src/dftNear2dZ.o     \
+../src/dftNear3d.o      \
+../src/calcNear3d.o     \
+../src/nearfield_c.o    \
+../src/zfeed.o          \
+../src/spara.o          \
+../src/outputZfeed.o    \
+../src/outputSpara.o    \
+../src/plot3dGeom.o     \
+../src/writeout.o       \
+../src/readout.o        \
+../src/post.o           \
+../src/farfield.o       \
+../src/outputFar0d.o    \
+../src/outputFar1d.o    \
+../src/outputFar2d.o    \
+../src/calcNear1d.o     \
+../src/calcNear2d.o     \
+../src/logFar1d.o       \
+../src/logFar2d.o       \
+../src/logNear1d.o      \
+../src/logNear2d.o      \
+../src/plot2dIter.o     \
+../src/plot2dFreq.o     \
+../src/plot2dFreq0.o    \
+../src/plot2dFar1d.o    \
+../src/plot2dFar1d0.o   \
+../src/plot3dFar2d.o    \
+../src/plot3dFar2d0.o   \
+../src/plot2dNear1d.o   \
+../src/plot2dNear1d0.o  \
+../src/plot2d3dNear2d.o \
+../src/plot2dNear2d0.o  \
+../src/plot3dNear2d0.o  \
+../src/plotTools.o      \
+../src/smithChart.o     \
+../src/ev2d.o           \
+../src/ev3d.o
+
+CC      = mpicc
+CFLAGS  = -O3 -I../include -Wall
+CFLAGS  += -std=c99
+LDFLAGS = -O3 $(OMPOPT)
+LIBS    = -lm
 
 $(PROGRAM): $(OBJS)
 	@echo "Loading $(PROGRAM) ... "
-	@$(CC) $(LDFLAGS) /Fe$(PROGRAM) $(OBJS) $(LIBS)
+	@$(CC) $(LDFLAGS) -o $(PROGRAM) $(OBJS) $(LIBS)
 
-Main.obj:
-	$(CC) /c $(CFLAGS) $(OMPOPT) $?
-comm.obj:
-	$(CC) /c $(CFLAGS) $(MPIOPT) $?
+Main.o: Main.c
+	$(CC) -c $(CFLAGS) $(OMPOPT) $?
+comm.o: comm.c
+	$(CC) -c $(CFLAGS) $(MPIOPT) $?
 
-.c.obj:
-	$(CC) /c $(CFLAGS) $<
+.c.o:
+	$(CC) -c $(CFLAGS) $<
 
 clean:
-	del *.obj
+	rm -f *.o
diff -ruN OpenFDTD.org/src/Makefile OpenFDTD/src/Makefile
--- OpenFDTD.org/src/Makefile	2019-06-08 12:53:12.000000000 +0900
+++ OpenFDTD/src/Makefile	2019-08-09 11:29:16.762581710 +0900
@@ -1,112 +1,114 @@
-PROGRAM = ../ofd.exe
-
-OMPOPT  = /openmp
-
-OBJS    =          \
-Main.obj           \
-input.obj          \
-monitor.obj        \
-memallocfree1.obj  \
-memallocfree2.obj  \
-memallocfree3.obj  \
-setup.obj          \
-setupSize.obj      \
-setupSizeNear.obj  \
-setupId.obj        \
-setupMurHx.obj     \
-setupMurHy.obj     \
-setupMurHz.obj     \
-setupPmlEx.obj     \
-setupPmlEy.obj     \
-setupPmlEz.obj     \
-setupPmlHx.obj     \
-setupPmlHy.obj     \
-setupPmlHz.obj     \
-setupPml.obj       \
-setupNear.obj      \
-solve.obj          \
-initfield.obj      \
-updateEx.obj       \
-updateEy.obj       \
-updateEz.obj       \
-updateHx.obj       \
-updateHy.obj       \
-updateHz.obj       \
-vfeed.obj          \
-efeed.obj          \
-epoint.obj         \
-murHx.obj          \
-murHy.obj          \
-murHz.obj          \
-pmlEx.obj          \
-pmlEy.obj          \
-pmlEz.obj          \
-pmlHx.obj          \
-pmlHy.obj          \
-pmlHz.obj          \
-dftNear1d.obj      \
-dftNear1dX.obj     \
-dftNear1dY.obj     \
-dftNear1dZ.obj     \
-dftNear2d.obj      \
-dftNear2dX.obj     \
-dftNear2dY.obj     \
-dftNear2dZ.obj     \
-dftNear3d.obj      \
-calcNear3d.obj     \
-nearfield_c.obj    \
-average.obj        \
-zfeed.obj          \
-spara.obj          \
-outputZfeed.obj    \
-outputSpara.obj    \
-plot3dGeom.obj     \
-writeout.obj       \
-readout.obj        \
-post.obj           \
-farfield.obj       \
-outputFar0d.obj    \
-outputFar1d.obj    \
-outputFar2d.obj    \
-calcFar1d.obj      \
-calcFar2d.obj      \
-calcNear1d.obj     \
-calcNear2d.obj     \
-logFar1d.obj       \
-logFar2d.obj       \
-logNear1d.obj      \
-logNear2d.obj      \
-plot2dIter.obj     \
-plot2dFreq.obj     \
-plot2dFreq0.obj    \
-plot2dFar1d.obj    \
-plot2dFar1d0.obj   \
-plot3dFar2d.obj    \
-plot3dFar2d0.obj   \
-plot2dNear1d.obj   \
-plot2dNear1d0.obj  \
-plot2d3dNear2d.obj \
-plot2dNear2d0.obj  \
-plot3dNear2d0.obj  \
-plotTools.obj      \
-smithChart.obj     \
-cputime.obj        \
-ev2d.obj           \
-ev3d.obj
-
-CC      = cl
-CFLAGS  = /Ox /nologo /I../include /W4 /wd4996
-CFLAGS  = $(CFLAGS) $(OMPOPT)
-#CFLAGS  = $(CFLAGS) /D__NEC__
-LDFLAGS = /Ox /nologo
-LIBS    =
-
-$(PROGRAM): $(OBJS)
-	@echo "Loading $(PROGRAM) ... "
-	@$(CC) $(LDFLAGS) /Fe$(PROGRAM) $(OBJS) $(LIBS)
-
-.c.obj:
-	$(CC) /c $(CFLAGS) $<
-
-clean:
-	del *.obj
+PROGRAM = ../ofd
+
+OMPOPT = -fopenmp
+
+OBJS    =        \
+Main.o           \
+input.o          \
+monitor.o        \
+memallocfree1.o  \
+memallocfree2.o  \
+memallocfree3.o  \
+setup.o          \
+setupSize.o      \
+setupSizeNear.o  \
+setupId.o        \
+setupMurHx.o     \
+setupMurHy.o     \
+setupMurHz.o     \
+setupPmlEx.o     \
+setupPmlEy.o     \
+setupPmlEz.o     \
+setupPmlHx.o     \
+setupPmlHy.o     \
+setupPmlHz.o     \
+setupPml.o       \
+setupNear.o      \
+solve.o          \
+initfield.o      \
+updateEx.o       \
+updateEy.o       \
+updateEz.o       \
+updateHx.o       \
+updateHy.o       \
+updateHz.o       \
+vfeed.o          \
+efeed.o          \
+epoint.o         \
+murHx.o          \
+murHy.o          \
+murHz.o          \
+pmlEx.o          \
+pmlEy.o          \
+pmlEz.o          \
+pmlHx.o          \
+pmlHy.o          \
+pmlHz.o          \
+dftNear1d.o      \
+dftNear1dX.o     \
+dftNear1dY.o     \
+dftNear1dZ.o     \
+dftNear2d.o      \
+dftNear2dX.o     \
+dftNear2dY.o     \
+dftNear2dZ.o     \
+dftNear3d.o      \
+calcNear3d.o     \
+nearfield_c.o    \
+average.o        \
+zfeed.o          \
+spara.o          \
+outputZfeed.o    \
+outputSpara.o    \
+plot3dGeom.o     \
+writeout.o       \
+readout.o        \
+post.o           \
+farfield.o       \
+outputFar0d.o    \
+outputFar1d.o    \
+outputFar2d.o    \
+calcFar1d.o      \
+calcFar2d.o      \
+calcNear1d.o     \
+calcNear2d.o     \
+logFar1d.o       \
+logFar2d.o       \
+logNear1d.o      \
+logNear2d.o      \
+plot2dIter.o     \
+plot2dFreq.o     \
+plot2dFreq0.o    \
+plot2dFar1d.o    \
+plot2dFar1d0.o   \
+plot3dFar2d.o    \
+plot3dFar2d0.o   \
+plot2dNear1d.o   \
+plot2dNear1d0.o  \
+plot2d3dNear2d.o \
+plot2dNear2d0.o  \
+plot3dNear2d0.o  \
+plotTools.o      \
+smithChart.o     \
+cputime.o        \
+ev2d.o           \
+ev3d.o
+
+CC      = gcc
+CFLAGS  = -O3 -I../include -Wall $(OMPOPT)
+CFLAGS  += -std=c99
+LDFLAGS = -O3 $(OMPOPT)
+LIBS    = -lm
+
+$(PROGRAM): $(OBJS)
+	@echo "Loading $(PROGRAM) ... "
+	@$(CC) $(LDFLAGS) -o $(PROGRAM) $(OBJS) $(LIBS)
+
+cputime.o: cputime.c
+	$(CC) -c -O2 $?
+
+.c.o:
+	$(CC) -c $(CFLAGS) $<
+
+clean:
+	rm -f *.o
